### PR TITLE
Foreign chain ballot details

### PR DESCRIPTION
**Pull request by nxtlvlrob**
Added two functions 
- `_getForeignBallotInfo` - To get the details from a foreign ballotBox
- `getTxReciept` - helper function to check transaction reciepts

### CODE REVIEW

File: src/light.ts [hunk 1]
```diff
@@ -202,21 +203,44 @@ type GetDemocNthBallot = t.TypeOf<typeof GetDemocNthBallotRT>
...
+    const { web3 } = svNetwork
+    const bbFarmContract = new web3.eth.Contract(BBFarmAbi, bbFarmAddress)
+    const votingNetworkDetails = await bbFarmContract.methods.getVotingNetworkDetails().call()
+    const { networkId, chainId, remoteBBFarmAddress } = explodeForeignNetDetails(votingNetworkDetails)
+
+    const { httpProvider } = svConst.getNetwork(networkId, chainId)
+
+    const remoteWeb3 = new Web3(httpProvider)
+    const remoteBBFarmContract = new remoteWeb3.eth.Contract(BBFarmAbi, remoteBBFarmAddress)
+    return await remoteBBFarmContract.methods.getDetails(ballotId, userEthAddress).call()
```
Comment (XertroV) [inline src/light.ts]: do we have a way to handle errors here?

File: src/light.ts [hunk 2]
```diff
@@ -498,6 +524,7 @@ export const submitEd25519Delegation = async (
     dlgtRequest: Bytes32,
     stellarPK: string,
     _signature: HexString,
+    networkName: string,
```
[the remark this answers is not in the document]
Comment (nxtlvlrob) [reply src/light.ts]: At this point it's just 'mainnet' or 'testnet' which is in the swarm-io-governance component docs

[the remark this answers is not in the document]
Comment (XertroV) [reply src/light.ts]: We should probs have an enum, not a priority atm. 

Either that or predicate on a type.

File: src/light.ts (hunk 1 above)
Comment (XertroV) [reply src/light.ts]: Alternative: does it matter if this errors?

File: src/light.ts [hunk 3]
```diff
@@ -225,10 +226,11 @@ export const getDemocNthBallot = async (svNetwork: SvNetwork, democBallotInfo: G
 
     const rawBallotSpecString = await getBallotSpec(archiveUrl, ethBallotDetails.specHash)
 
+
     return { ...bbFarmAndBallotId, ...ethBallotDetails, rawBallotSpecString, data: JSON.parse(rawBallotSpecString) }
 }
 
-const _getForeignBallotInfo = async (svNetwork: SvNetwork, ballotId: string, bbFarmAddress: EthAddress, userEthAddress: EthAddress) => {
+const _getForeignBallotInfo = async (svNetwork: SvNetwork, ballotId: string, bbFarmAddress: EthAddress, userEthAddress: EthAddress): Promise<any> => {
```
Comment (XertroV) [inline src/light.ts]: :P This is a little pointless; we'll fix it up in future though

File: src/light.ts [hunk 4]
```diff
@@ -554,7 +555,7 @@ export const publishSignedTx = async (web3: any, rawTx: string): Promise<string>
     return await web3.eth.sendSignedTransaction(rawTx).then(r => r.transactionHash)
 }
 
-export const getTxReciept = async (netConf: EthNetConf, txId: Bytes32): Promise<string> => {
+export const getTxReciept = async (netConf: EthNetConf, txId: Bytes32): Promise<object> => {
```
Comment (XertroV) [inline src/light.ts]: also a bit pointless, no worries tho (actually, this might break some type signature stuff, but if so we'll fix later)

**Pull request merged**